tangelo/linq/tests/test_translator_perf.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2023 Good Chemistry Company.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" Test class to keep track of performance of translator module, for format convertion """
+
+import unittest
+import os
+import time
+
+from itertools import product
+
+from tangelo.linq import Gate, Circuit
+from tangelo.toolboxes.operators import QubitOperator
+from tangelo.helpers.utils import installed_backends
+from tangelo.linq import translate_operator, translate_circuit
+from tangelo.linq.translator.translate_qubitop import FROM_TANGELO as FROM_TANGELO_OP
+from tangelo.linq.translator.translate_qubitop import TO_TANGELO as TO_TANGELO_OP
+from tangelo.linq.translator.translate_circuit import FROM_TANGELO as FROM_TANGELO_C
+from tangelo.linq.translator.translate_circuit import TO_TANGELO as TO_TANGELO_C
+
+
+# Build artificially large operator made of all possible "full" Pauli words (no 'I') of length n_qubits
+n_qubits_op = 10
+n_terms = 3 ** n_qubits_op
+terms = {tuple(zip(range(n_qubits_op), pw)): 1.0 for pw in product(['X', 'Y', 'Z'], repeat=n_qubits_op)}
+tangelo_op = QubitOperator()
+tangelo_op.terms = terms
+
+# Build artificially large quantum circuit
+n_qubits = 20
+n_repeat = 4000
+gates = [Gate('X', i) for i in range(n_qubits)] + [Gate("CNOT", (i+1) % n_qubits, control=i) for i in range(n_qubits)]
+tangelo_c = Circuit(gates * n_repeat)
+
+
+class PerfTranslatorTest(unittest.TestCase):
+
+    def test_perf_operator(self):
+        """ Performance test with a reasonable large input for operator """
+
+        print(f'\n[Performance Test :: linq operator format conversion]')
+        print(f'\tInput size: n_qubits={n_qubits_op}, n_terms={n_terms}\n')
+
+        for f in FROM_TANGELO_OP:
```

Comment:
Right now:

- try / except blocks are here because we don't care about errors (we already have numerical tests somewhere else) and to ignore any issue arising from backends not being installed. e.g it will try to run everything it finds
- currently only try to translate to tangelo for whatever is supported the other way.

For operator, this covers everything. For circuits, this does not cover the very few formats whose translation is only supported to tangelo, but it does not seem like anything users have cared about so far (also, supporting them means creating the input circuit in their format and either reading from file or a variable) and we know we do not use backends relying on these formats yet.